lib/os/reboot.c

```diff
@@ -14,7 +14,10 @@ extern void sys_arch_reboot(int type);
 FUNC_NORETURN void sys_reboot(int type)
 {
 	(void)irq_lock();
-	sys_clock_disable();
+
+	if (IS_ENABLED(CONFIG_SYSTEM_TIMER_HAS_DISABLE_SUPPORT)) {
+		sys_clock_disable();
+	}
```

Comment:
I still don't see why this is better than the previous code.

Reply:
It's subjective, but this looks much cleaner to me than having an empty stub in a header file.

Reply:
Actually, if we're consistent, we should do either provide stubs or:
https://github.com/zephyrproject-rtos/zephyr/blob/abe50cc92991389e2508de8ea721bee36cd5fb17/include/zephyr/drivers/gpio.h#L864

that is, not provide the declaration if not supported. That also means `IS_ENABLED` can't be used.

Reply:
> Actually, if we're consistent, we should do either provide stubs or:
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/abe50cc92991389e2508de8ea721bee36cd5fb17/include/zephyr/drivers/gpio.h#L864
> 
> that is, not provide the declaration if not supported. That also means `IS_ENABLED` can't be used.

Actually no, approach in GPIO results in a link error, since only the syscall is guarded. Meaning `IS_ENABLED` can also be used.